src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/KeyworkerServiceController.kt

```diff
@@ -0,0 +1,291 @@
+    @GetMapping(path = ["/prison/{prisonId}"])
+    fun getPrisonMigrationStatus(@ApiParam("prisonId") @PathVariable("prisonId") prisonId: String): Prison {
+        val prisonDetail = prisonSupportedService.getPrisonDetail(prisonId)
+        return prisonDetail ?: Prison.builder().prisonId(prisonId).supported(false).build()
+    }
+
+    @ApiOperation(value = "Enable Manual Allocation and Migrate", notes = "Role Required: KW_MIGRATION. This will invoke migration from NOMIS DB")
+    @ApiResponses(value = [ApiResponse(code = 200, message = "OK", response = Prison::class), ApiResponse(code = 400, message = "Invalid request.", response = ErrorResponse::class), ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse::class)])
+    @PostMapping(path = ["/enable/{prisonId}/manual"])
+    fun addSupportedPrisonForManualAllocation(
+        @ApiParam("prisonId") @PathVariable("prisonId") prisonId: String,
+        @ApiParam("migrate") @Param("migrate") migrate: Boolean?,
+        @ApiParam(name = "capacity", value = "standard and extended default keyworker capacities for this prison, comma separated, e.g. &capacity=6,9") @Param("capacity") capacity: Array<Int>?,
+        @ApiParam(name = "frequency", value = "default KW Session Frequency in weeks (default 1)") @Param("capacity") frequency: Int?
+    ): Prison = updateAndMigrate(prisonId, migrate, false, capacity, frequency)
+
+    @ApiOperation(value = "Enable Auto Allocation for specified prison and Migrate", notes = "Role Required: KW_MIGRATION. This will also invoke migration from NOMIS DB")
+    @ApiResponses(value = [ApiResponse(code = 200, message = "OK", response = Prison::class), ApiResponse(code = 400, message = "Invalid request.", response = ErrorResponse::class), ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse::class)])
+    @PostMapping(path = ["/enable/{prisonId}/auto-allocate"])
+    fun addSupportedPrisonForAutoAllocation(
+        @ApiParam("prisonId") @PathVariable("prisonId") prisonId: String,
+        @ApiParam("migrate") @Param("migrate") migrate: Boolean?,
+        @ApiParam(name = "capacity", value = "standard and extended default keyworker capacities for this prison, comma separated, e.g. &capacity=6,9") @Param("capacity") capacity: Array<Int>?,
+        @ApiParam(name = "frequency", value = "default KW Session Frequency in weeks (default 1)") @Param("capacity") frequency: Int?
+    ): Prison {
+        return updateAndMigrate(prisonId, migrate, true, capacity, frequency)
+    }
+
+    private fun updateAndMigrate(prisonId: String?, migrate: Boolean? = false, autoAllocate: Boolean, capacity: Array<Int>?, kwSessionFrequencyInWeeks: Int? = 1): Prison {
```

Comment:
have you got your nullability right here - why is `prisonId` allowed to be `null` if it is a path parameter?  Why supply a default to `migrate` when it is always passed as a parameter?

Reply:
agree re `prisonId`
regarding `migrate` it's optional from the endpoint method but can default it in the controller method before passing it in